src/main/java/org/openrewrite/staticanalysis/NeedBraces.java

```diff
@@ -140,7 +167,15 @@ public J.If visitIf(J.If iff, ExecutionContext ctx) {
             J.If elem = super.visitIf(iff, ctx);
             boolean hasAllowableBodyType = elem.getThenPart() instanceof J.Block;
             if (!needBracesStyle.getAllowSingleLineStatement() && !hasAllowableBodyType) {
-                J.Block b = buildBlock(elem, elem.getThenPart());
+                J.Block b;
+                if (elem.getElsePart() != null && !elem.getElsePart().getPrefix().getComments().isEmpty()) {
+                    Space end = elem.getElsePart().getPrefix();
+                    elem = elem.withElsePart(elem.getElsePart().withPrefix(Space.EMPTY));
+                    b = buildBlock(elem.getThenPart()).withEnd(end);
+                } else {
+                    b = buildBlock(elem.getThenPart());
+                }
+
```

Comment:
trailing comments for if statements can appear as a prefix in the `else`

example:
```
static void method() {
    if (true) return; // if comment
    else return; // else comment
}
```